core/trino-spi/src/main/java/io/trino/spi/type/TypeOperators.java

```diff
@@ -538,7 +544,7 @@ private static int getScore(OperatorMethodHandle operatorMethodHandle)
             if (argument == NULL_FLAG || argument == FLAT) {
                 score += 100;
             }
-            else if (argument == BLOCK_POSITION) {
+            else if (argument == BLOCK_POSITION || argument == VALUE_BLOCK_POSITION_NOT_NULL) {
```

Comment:
Why is this needed? Have you checked if it has any impact on other usages not related to ordering?